cli/azd/internal/cmd/add/add_configure.go

```diff
@@ -58,6 +58,16 @@ func Configure(
 		return r, nil
 	case project.ResourceTypeStorage:
 		return fillStorageDetails(ctx, r, console, p)
+	case project.ResourceTypeKeyVault:
+		if _, exists := p.PrjConfig.Resources["vault"]; exists {
+			return nil, fmt.Errorf(
+				"you already have a project key vault named 'vault'. " +
+					"To add a secret to it, run 'azd env set-secret <name>'",
+			)
+		}
```

Comment:
`project key vault` ?  Why do we call it a project?

If we are trying to limit having just one only key vault, should we check for the type?  I think folks can manually add a key vault resource with a name like `kv-foo` and this condition wouldn't catch it

Reply:
> `project key vault` ? Why do we call it a project?

That's what we landed on in our initial designs, but open to calling it something else

> If we are trying to limit having just one only key vault, should we check for the type? I think folks can manually add a key vault resource with a name like kv-foo and this condition wouldn't catch it

Yes that's a limitation of resource types in composability that support at most 1 resource today. Similar checks exist for other resource types:

![image](https://github.com/user-attachments/assets/ac59a373-c01b-4cc0-95d6-8ba5467d07cb)

Want to add anything here @weikanglim?

Reply:
FWIW, @JeffreyCA, we do have an extra safeguard in `scaffold_gen.go` for `infra synth` to generate an error for multiple resources of a particular type. But here, the idea is presented along the lines of "there is one primary azd resource called `vault`" and that is special.

I'm not sure if it's helpful to orient ourselves along the lines that we're trying to simplify a building experience for a new user to Azure.

In these scenarios, the user doesn't really need more than one of anything -- they really just see Azure as a cloud provider that allows them to use different types of functionality.